File: mobile/src/services/database/models/OneRepMax.ts
import { Model } from '@nozbe/watermelondb';
import { field, date, readonly } from '@nozbe/watermelondb/decorators';

export class OneRepMax extends Model {
  static table = 'one_rep_maxes';

  @field('user_id') userId!: string;
  @field('exercise_name') exerciseName!: string;
  @field('weight') weight!: number;
  @field('date') date!: number;
  @readonly @date('created_at') createdAt!: Date;
  @readonly @date('updated_at') updatedAt!: Date;
  @date('synced_at') syncedAt?: Date;
}

File: backend/internal/domain/one_rep_max.go
package domain

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// OneRepMax represents a one-rep max record for an exercise
type OneRepMax struct {
	ID           uuid.UUID      `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	UserID       uuid.UUID      `gorm:"type:uuid;not null;index:idx_one_rep_maxes_user_exercise" json:"user_id"`
	ExerciseName string         `gorm:"type:varchar(255);not null;index:idx_one_rep_maxes_user_exercise" json:"exercise_name"`
	Weight       float64        `gorm:"type:decimal(6,2);not null" json:"weight"`
	Date         time.Time      `gorm:"type:date;not null;index:idx_one_rep_maxes_date" json:"date"`
	CreatedAt    time.Time      `gorm:"not null;default:CURRENT_TIMESTAMP" json:"created_at"`
	UpdatedAt    time.Time      `gorm:"not null;default:CURRENT_TIMESTAMP" json:"updated_at"`
	DeletedAt    gorm.DeletedAt `gorm:"index" json:"-"`

	// Relationships
	User User `gorm:"foreignKey:UserID" json:"user,omitempty"`
}

// TableName specifies the table name for OneRepMax model
func (OneRepMax) TableName() string {
	return "one_rep_maxes"
}

// BeforeCreate hook to set UUID if not provided
func (o *OneRepMax) BeforeCreate(tx *gorm.DB) error {
	if o.ID == uuid.Nil {
		o.ID = uuid.New()
	}
	return nil
}
